src/genie/libs/parser/iosxe/show_wlan.py

```diff
@@ -0,0 +1,106 @@
+import re
+
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Any, Optional
+
+
+# ======================
+# Schema for:
+#  * 'show wlan summary'
+# ======================
+class ShowWlanSummarySchema(MetaParser):
+    """Schema for show wlan summary."""
+
+    schema = {
+        "wlan_summary": {
+            "wlan_count": int,
+            "wlan_id": {
+                int: {
+                    "profile_name": str,
+                    "ssid": str,
+                    "wlan_status": str,
+                    "status_security": str
+                }
+            }
+        }
+    }
+
+
+# ======================
+# Parser for:
+#  * 'show wlan summary'
+# ======================
+class ShowWlanSummary(ShowWlanSummarySchema):
+    """Parser for show wlan summary"""
+
+    cli_command = 'show ap cdp neighbor'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        wlan_summary_dict = {}
+
+        # Number of WLANs: 4
+        #
+        # ID   Profile Name                     SSID                             Status Security
+        # ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        # 17   lizzard_Global                  lizzard                         UP     [WPA2][802.1x][FT + 802.1x][AES],[FT Enabled]
+        # 18   wip_Global                      wip                             UP     [WPA2][802.1x + CCKM][AES]
+        # 19   internet_Global                  internet                         UP     [open],MAC Filtering
+        # 20   lizzard-l_Global                lizzard-legacy                  UP     [WPA2][802.1x][AES]
+
+        # Number of WLANs: 4
+        wlan_count_capture = re.compile(r"^Number\s+of\s+WLANs:\s+(?P<wlan_count>\d+)$")
+        # ID   Profile Name                     SSID                             Status Security
+        wlan_info_header_capture = re.compile(r"^ID\s+Profile\s+Name\s+SSID\s+Status\s+Security$")
+        # ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        delimiter_capture = re.compile(
+            r"^----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------$")
```

Comment:
This is not required, if a line doesn't match any regex then it will be skipped (which is the same as matching and doing nothing). Please remove.

Reply:
I disagree with this design pattern, I actually think this leads to more issues over time. The positive affirmation of the project will cause issues in the long run. I took this from FAQ on ntc-templates (which I help maintain) 

```
Why is there a requirement to use Error in every template?

Ensuring that the textfsm template can account for every line is the only method to ensure that data was not accidentally missed. Take the following example. Initially we account for status to be:

Value STATUS (up|down)

Given the result of:

Interface                      Status         Protocol Description
Gi0/0/1                        admin down     down
Gi0/0/2                        up             up       ISP Connection
Gi0/0/3                        down           down
The output would miss the G0/0/1 interface, since the STATUS of admin down not known. If this was a low percentage use case, it can go undetected, and result in incorrect information being returned. Instead, by ensuring that we fail fast, an Error is raised and hopefully GitHub Issue is put in.
```

That being said, it is up to you, please respond either way.

Reply:
I fully agree with @itdependsnetworks. This parsing strategy makes the pyats parsing highly unreliable, especially for commands like show mac address-table, where some lines can be missed and is almost impossible to detect.

Reply:
@itdependsnetworks I see what you mean. As of current we don't raise an error if lines are not parsed, but it's a good idea that I've mentioned to the team. Keep the current implementation you have as it wont hurt.